### DISCUSSION TITLE
Testing error "NextRouter was not mounted."

### DISCUSSION THREAD

**Discussion by thomasweigert91** [neuefische/questions#338] [Web] [answered]
Hey,

I am getting the following error in my first test. No idea, what I am doing wrong. Everything in my Navigation works fine.
I tried different tests from the other students, but I am getting the same error every time.

 _FAIL  __tests__/index.test.js
  Home
    ✕ checks if there is a nav component in HomePage (125 ms)

  ● Home › checks if there is a nav component in HomePage

    Error: NextRouter was not mounted. https://nextjs.org/docs/messages/next-router-not-mounted

      2 |
      3 | const ActiveLink = ({href, children}) => {
    > 4 |   const router = useRouter();
        |                           ^
      5 |   const handleClick = e => {
      6 |     e.preventDefault();
      7 |     router.push(href);_

_____

This is my function:

import {useRouter} from "next/router";

const ActiveLink = ({href, children}) => {
  const router = useRouter();
  const handleClick = e => {
    e.preventDefault();
    router.push(href);
  };

  const isCurrentPath = router.pathname === href;

  return (
    <a
      href={href}
      onClick={handleClick}
      style={{color: isCurrentPath ? "#735CDD" : "#8A878E"}}
    >
      {children}
    </a>
  );
};

export default ActiveLink;

**Comment by actyralf** [comment]
This is a problem when using `useRouter()` outside of a Next.js application - which is true for component testing or Storybook likewise.
I'll prepare a workaround for this - for the time being, disable the failing test.

**Comment by actyralf** [comment, marked as answer]
Here is the workaround:

1. Install `next-router-mock` as a dev dependency (`npm i --save-dev next-router-mock`)
2. Add these lines after the import section of your `ComponentXY.test.js` file:
   ```
   import mockRouter from "next-router-mock";

   jest.mock("next/router", () => require("next-router-mock"));
   jest.mock("next/dist/client/router", () => require("next-router-mock"));
   ```
3. Before rendering your component in the test (`render(<ComponentXY/>)`), call this function:
   ```

   mockRouter.setCurrentUrl("/yourpath");
   ```

This allows you to test your component for different pathnames. You could test for example if your highlighting works by testing this:

```
const testLink = screen.getByRole("link", {
      name: /Test/i,
    });
    expect(testLink).toBeInTheDocument();
    expect(testLink).toHaveStyle("color:#735CDD");
```
